SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+; Startup type for Browser Service. 
+BROWSERSVCSTARTUPTYPE="Automatic"
+'@
+
+    # Check if there are designated drives for Data, Log, TempDB, Back-up and Application.
+    If ($DataVolume -eq $null -or $DataVolume -eq '') {
+        $DataVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne  $_.DriveLetter -and $_.FileSystemLabel -like '*Data*'} | 
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($LogVolume -eq $null -or $LogVolume -eq '') {
+        $LogVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Log*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($TempVolume -eq $null -or $TempVolume -eq '') {
+        $TempVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*TempDB*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($AppVolume -eq $null -or $AppVolume -eq '') {
+        $AppVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*App*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($BackupVolume -eq $null -or $BackupVolume -eq '') {
+        $BackupVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Backup*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    #Check the number of cores available on the server. Summed because every processor can contain multiple cores
+    $NumberOfCores = Get-WmiObject -Class Win32_processor |  
+        Measure-Object NumberOfLogicalProcessors -Sum | 
+        Select-Object -ExpandProperty sum
+
+    IF ($NumberOfCores -gt 8)
+    { $NumberOfCores = 8 }
+
+    #Get the amount of available memory. If it's more than 40 GB, give the server 10% of the memory, else reserve 4 GB.
+
+    $ServerMemory = Get-WmiObject -Class win32_physicalmemory | 
+        Measure-Object Capacity -sum | 
+        Select-Object -ExpandProperty sum
+    $ServerMemoryMB = ($ServerMemory / 1024) / 1024
+
+    If ($ServerMemoryMB -gt 40960) {
+        $ServerWinMemory = $ServerMemoryMB * 0.1
+        $ServerMemoryMB = $ServerMemoryMB - $ServerWinMemory
+    }
+    else {
+        $ServerMemoryMB = $ServerMemoryMB - 4096
+    }
+
+    IF ($null -eq $DataVolume -or $DataVolume -eq '') {
+        $DataVolume = 'C'
+    }
+
+    IF ($null -eq $LogVolume -or $LogVolume -eq '') {
+        $LogVolume = $DataVolume
+    }
+
+    IF ( $null -eq $TempVolume -or $TempVolume -eq '') {
+        $TempVolume = $DataVolume
+    }
+
+    IF ( $null -eq $AppVolume -or $AppVolume -eq '') {
+        $AppVolume = 'C'
+    }
+
+    IF ( $null -eq $BackupVolume -or $BackupVolume -eq '') {
+        $BackupVolume = $DataVolume
+    }
+
+
+
+    Clear-Host
```

Comment:
dbatools does not use any -Host commands. Generally, we try to output objects to the user.

Reply:
command removed. This was a left over from the basics when it was just a ps script.